db/data_migrations/scrub_database.rb

```diff
@@ -1,17 +1,45 @@
+require 'i18n'
+# I18n.load_path += Dir[File.join(File.expand_path(File.dirname(__FILE__)), 'faker/locales', '*.yml')]
+require 'faker'
+# I18n.reload!
+# Faker::Config.locale = 'en'
+
 class ScrubDatabase
+  I18n.config.available_locales = %i[en de]
+  I18n.default_locale = :en # prime savior of the faker gem
+  
+  
+  I18n.reload!
+
+  
+
   def self.run!
+    duplisArray = []
     User.find_each do |user|
-      user.name = "#{user.role.name.downcase.gsub(/[- ]/, '_')}#{user.id}"
-      user.fullname = "#{user.id}, #{user.role.name.downcase.gsub(/[- ]/, '_')}"
-      user.email = 'expertiza@mailinator.com'
-      user.handle = 'handle'
-      user.password = 'password'
-      # user.password_confirmation = "password"
-      user.save(validate: false)
+      loop do
+        fake_name = Faker::Name.first_name
+        fake_lastname = Faker::Name.last_name
+        num = rand(1..9)
+        role = user.role.name.downcase.gsub(/[- ]/, '_')
+        user.name = role == 'student' ? "#{fake_name}_#{fake_lastname[0..3]}#{num}" : "#{role}_#{fake_name}__#{fake_lastname[0..3]}"
+        user.fullname = role == 'student' ? "#{fake_lastname}, #{fake_name}" : "#{fake_name}, #{role}"
+        user.email = 'expertiza@mailinator.com'
+        user.handle = 'handle'
+        user.password = 'password'
+        username = role == 'student' ? "#{fake_name}_#{fake_lastname[0..3]}#{num}" : "#{role}_#{fake_name}__#{fake_lastname[0..3]}"
+        # Checking if username already exists
+        if User.find_by(:name => username) == nil #no record exists of that username
```

Comment:
For each user, the loop ensures that a unique username is generated, and once generated, only then it would exit out of the loop; else it will stay in the loop

Reply:
The Faker::Name.first_name does not guarantee unique name generation. Although unlikely, couldn't two users end up with the same name ?

Reply:
As pointed out by Prasad, there are two loops and the inner loop is only broken if there is no collision of names.